docs/pages/ktable.vue

```diff
@@ -0,0 +1,173 @@
+<template>
+
+  <DocsPageTemplate apiDocs>
+    <DocsPageSection title="Overview" anchor="#overview">
+      <p>
+        The <code>KTable</code> component is an accessible and customizable table component designed to handle a variety of data presentation needs. The component is suitable for both simple and complex data tables. It offers:
+        <ul>
+          <li>Built-in sorting as well as integration with externally sorted data</li>
+          <li>Keyboard navigation</li>
+          <li>Dynamic column resizing</li>
+          <li>Sticky headers</li>
+        </ul>
+      </p>
+    </DocsPageSection>
+    <DocsPageSection title="Usage" anchor="#usage">
+      <!--Non-Sortable Table-->
+      <h3>Table without sorting functionality</h3>
+      <p>
+        This is an example to show how <code>KTable</code> can be used without any sorting functionality, for a simple table.
+      </p>
+      <!-- eslint-disable -->
+      <DocsShowCode language="html">
+        <KTable
+          :headers="headers"
+          :rows="rows"
+          caption="Non Sortable Table"
+          :sortable="false"
+        >
+
+          <template #header="{ header, index }">
+            <span>{ header.label } (Backend)</span>
+          </template>
+          <template #cell="{ content, rowIndex, colIndex }">
+            <span v-if="colIndex === 2">{ content } (City)</span>
```

Comment:
Feel fee to brainstorm together, just chiming in to mark as a possible _follow-up_.
 
(this could indeed be useful @AlexVelezLl, however with @BabyElias we agreed to only focus on high-level public API adjustments, and even those are _not required_ - the table was originally meant to be just a prototype :)

Reply:
Feel fee to brainstorm together, just chiming in to mark as a possible _follow-up_.
 
(this could indeed be useful @AlexVelezLl, however with @BabyElias we agreed to only focus on high-level public API adjustments, and even those are _not required_ - the table was originally meant to be just a prototype :)

Reply:
Feel fee to brainstorm together, just chiming in to mark as a _follow-up_.
 
(this could indeed be useful @AlexVelezLl, however with @BabyElias we agreed to only focus on high-level public API adjustments, and even those are not required - the table was originally meant to be just a prototype :)

Reply:
That's not to dismiss the comment, it's very good! Just doing a bit of triaging here.

Reply:
Yes, yes! This could be a future improvement. I can do a follow up proposal issue if you agree with this

Reply:
Oh lovely, thanks so much for your offer! I am fine chiming in later on if you'd like to hear on something particular, but for now will let @BabyElias to enjoy the discussion and make some conclusions together with you if she'd like to and there's time :) (but no pressure @BabyElias, no problem to categorize this as post-GSOC)

Reply:
That sounds like a good addition to KTable, thanks for suggesting this @AlexVelezLl !
I didn't have an idea about dynamic slots before this, read about them now for the first time. I agree that they will be more useful than the existing slot arrangement. However, I am not sure if I can cover them during the GSoC period or not. As suggested by @MisRob, I guess we can add it to triage/future enhancements and then maybe I can look into it myself post GSoC as well :)

Reply:
Perfect! I'm gonna create the issue just to immortalize the idea, and we can discuss it later on, no pressure to look at it for now :hugs:.